hardware/MQTTAutoDiscover.cpp

```diff
@@ -3467,6 +3475,24 @@ void MQTTAutoDiscover::handle_auto_discovery_climate(_tMQTTASensor* pSensor, con
 		bIsJSON = root.isObject();
 	}
 
+	// Check for legacy devices once on first message
+	if (pSensor->devType == 0)
+	{
+		// If both topics exist, check database for old devices
+		if (!pSensor->temperature_command_topic.empty() && !pSensor->current_temperature_topic.empty())
+		{
+			std::vector<std::vector<std::string>> old_dev;
+			old_dev = m_sql.safe_query("SELECT COUNT(*) FROM DeviceStatus WHERE (HardwareID==%d) AND (DeviceID=='%q') AND (Unit==%d) AND (Type==%d OR Type==%d)",
+				m_HwdID, pSensor->unique_id.c_str(), CLIMATE_TEMP_SETPOINT_UNIT, pTypeTEMP, pTypeSetpoint);
+			pSensor->bUseLegacyClimate = !old_dev.empty() && atoi(old_dev[0][0].c_str()) > 0;
```

Comment:
Does this mean that when a user already had 'any' old thermostats, it will never get to enjoy the new type?